iac/stacks/common.py

```diff
@@ -62,33 +92,12 @@ def __init__(
             subnets=[ec2.SubnetSelection(subnet_type=ec2.SubnetType.PRIVATE_WITH_EGRESS)],
         )
 
-        # create a secret so we can manually set the username
-        database_credentials_secret = sm.Secret(
-            self,
-            "DBCredentialsSecret",
-            secret_name="common/mermaid-db/creds",
-            generate_secret_string=sm.SecretStringGenerator(
-                secret_string_template=json.dumps({"username": "mermaid_admin"}),
-                generate_string_key="password",
-                exclude_punctuation=True,
-                include_space=False,
-            ),
+        self.dev_database = self._database(
+            id="DevDatabase", version=rds.PostgresEngineVersion.VER_16_3
         )
 
-        self.database = rds.DatabaseInstance(
-            self,
-            "PostgresRdsV2",
-            vpc=self.vpc,
-            engine=rds.DatabaseInstanceEngine.postgres(version=rds.PostgresEngineVersion.VER_13_7),
-            instance_type=ec2.InstanceType.of(
-                ec2.InstanceClass.BURSTABLE3,
-                ec2.InstanceSize.SMALL,
-            ),
-            vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PRIVATE_ISOLATED),
-            backup_retention=Duration.days(7),
-            deletion_protection=True,
-            removal_policy=RemovalPolicy.SNAPSHOT,
-            credentials=rds.Credentials.from_secret(database_credentials_secret),
+        self.prod_database = self._database(
+            id="ProdDatabase", version=rds.PostgresEngineVersion.VER_13_7
```

Comment:
With CDK, when you change the name of the `id`, it will common delete the resource. If you look at the diff that ran in the GHA, there is a line that says `[-] AWS::RDS::DBInstance PostgresRdsV2 PostgresRdsV2B4B63A33 destroy`, which means it will delete the production database. If you keep the `id=PostgresRdsV2`, and don't change anything else, it won't get deleted.